recipes/scc/all/conanfile.py

```diff
@@ -1,89 +1,166 @@
 
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    # no exports_sources attribute, but export_sources(self) method instead
-    # this allows finer grain exportation of patches per version
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-            
+        export_conandata_patches(self)
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-            
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://github.com/Minres/SystemC-Components/blob/2023.06/src/sysc/scc/perf_estimator.h#L20
+        self.requires("boost/1.83.0", transitive_headers=True, transitive_libs=True)
+        # https://github.com/Minres/SystemC-Components/blob/2023.06/src/sysc/scc/trace/vcd_trace.hh#L28
+        self.requires("fmt/10.2.1", transitive_headers=True, transitive_libs=True)
+        # https://github.com/Minres/SystemC-Components/blob/2023.06/src/common/util/lz4_streambuf.h#L13
+        self.requires("lz4/1.9.4", transitive_headers=True, transitive_libs=True)
+        self.requires("rapidjson/cci.20220822")
+        self.requires("spdlog/1.13.0")
+        # https://github.com/Minres/SystemC-Components/blob/2023.06/src/sysc/tlm/scc/lwtr/tlm2_lwtr.h
+        self.requires("systemc-cci/1.0.0", transitive_headers=True, transitive_libs=True)
+        self.requires("systemc/2.3.4", transitive_headers=True, transitive_libs=True)
+        # https://github.com/Minres/SystemC-Components/blob/2023.06/src/sysc/scc/trace/gz_writer.hh#L18
+        self.requires("zlib/[>=1.2.11 <2]", transitive_headers=True, transitive_libs=True)
+        self.requires("yaml-cpp/0.8.0")
+
     def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(f"{self.name} is not supported on {self.settings.os}.")
+        if is_msvc(self):
+            # Fails with
+            #   src\sysc\tlm\scc\tlm_mm.h(116,114): error C2259: 'tlm::scc::tlm_gp_mm_t<16,false>': cannot instantiate abstract class
+            #     (compiling source file '../../../src/src/sysc/tlm/scc/lwtr/tlm2_lwtr.cpp')
+            #     src\sysc\tlm\scc\tlm_mm.h(116,48):
+            #     see declaration of 'tlm::scc::tlm_gp_mm_t<16,false>'
+            #     src\sysc\tlm\scc\tlm_mm.h(116,114):
+            #     due to following members:
+            #     src\sysc\tlm\scc\tlm_mm.h(116,114):
+            #     'void tlm::tlm_extension<tlm::scc::tlm_gp_mm>::copy_from(const tlm::tlm_extension_base &)': is abstract
+            #     systemc-2.3.4\p\include\tlm_core\tlm_2\tlm_generic_payload\tlm_gp.h(78,18):
+            #     see declaration of 'tlm::tlm_extension<tlm::scc::tlm_gp_mm>::copy_from'
+            # and
+            #   src\sysc\tlm\scc\tlm_mm.h(31,20): error C2061: syntax error: identifier '__attribute__'
+            raise ConanInvalidConfiguration(f"{self.ref} is not supported on MSVC.")
```

Comment:
Is this documented and explicit by the upstream library? The message should be different if the recipe does not support it yet, or it is a library documented limitation. If it is a recipe issue, then: "This recipe does not support MSVC yet, contributions to ConanCenter to add it are welcome"

Reply:
It's documented in the recipe.

Reply:
Other than that - your guess is as good as mine. I'll leave it up to the user to find the deeper reason for the failure.

Reply:
> It's documented in the recipe.

The problem is that end users of the package doesn't read the recipe, they just receive an exception saying "scc/version is not supported in MSVC", which could be understood as the upstream library does not support MSVC which doesn't seem true. Then, it is important for end recipe users (as opposed to recipe creators and contributors) to get the right message.

Reply:
Since the project appears to intend to have MSVC support, but I did not find any more details about it on the repo, I included a "contributions are welcome" message in the error.